test/topology_custom/test_mv_topology_change.py

```diff
@@ -77,3 +80,63 @@ async def do_writes(start_it, repeat) -> int:
     stop_event.set()
     await asyncio.gather(*tasks)
 
+# Reproduces #19152
+# Verify a pending replica is not doing unnecessary work of building and sending view updates.
+# 1) we have a table with a materialized view
+# 2) start migrating a tablet from node 1 to node 2
+# 3) while node 2 is a pending replica, write to the table
+# 4) complete migration
+# 5) verify node 2 did not build view updates
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_update_on_pending_replica(manager: ManagerClient):
+    cfg = {'enable_tablets': True}
+    servers = [await manager.server_add(config=cfg)]
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 1};")
+    await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
+    await cql.run_async("CREATE MATERIALIZED VIEW test.mv1 AS SELECT * FROM test.test WHERE pk IS NOT NULL AND c IS NOT NULL PRIMARY KEY (c, pk);")
+
+    servers.append(await manager.server_add(config=cfg))
+
+    key = 7
+    tablet_token = 0
```

Comment:
I had to look at other tests to see a comment `# Doesn't matter since there is one tablet` that explains this token business (this token doesn't fit the key). So maybe that comment is useful :-)